# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderCompositorANGLE.h
# Commit: 971767c6ed2b
# Full Hash: 971767c6ed2b2201e172105046ecc61942e03a7e
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2020-01-24 21:46:30
# Regressor Bug: 1602643
# File Overlap Count: 1
# Description:
#   Bug 1602643 - Disable WebRender compositor dinamically for async screenshot r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D59658
# ==============================================================================

diff -r 436df7dc06c6 -r 971767c6ed2b gfx/webrender_bindings/RenderCompositorANGLE.h
--- a/gfx/webrender_bindings/RenderCompositorANGLE.h	Fri Jan 24 01:04:09 2020 +0000
+++ b/gfx/webrender_bindings/RenderCompositorANGLE.h	Thu Jan 23 23:52:05 2020 +0000
@@ -64,6 +64,8 @@
 
   bool SurfaceOriginIsTopLeft() override { return true; }
 
+  bool SupportAsyncScreenshot() override;
+
   bool ShouldUseNativeCompositor() override;
   uint32_t GetMaxUpdateRects() override;
 
@@ -80,6 +82,7 @@
   void DestroyTile(wr::NativeSurfaceId aId, int32_t aX, int32_t aY) override;
   void AddSurface(wr::NativeSurfaceId aId, wr::DeviceIntPoint aPosition,
                   wr::DeviceIntRect aClipRect) override;
+  void EnableNativeCompositor(bool aEnable) override;
 
   // Interface for partial present
   bool UsePartialPresent() override;
@@ -95,7 +98,7 @@
   void InitializeUsePartialPresent();
   void InsertGraphicsCommandsFinishedWaitQuery(
       RenderedFrameId aRenderedFrameId);
-  bool WaitForPreviousGraphicsCommandsFinishedQuery();
+  bool WaitForPreviousGraphicsCommandsFinishedQuery(bool aWaitAll = false);
   bool ResizeBufferIfNeeded();
   bool CreateEGLSurface();
   void DestroyEGLSurface();
@@ -106,6 +109,7 @@
                                                   bool aUseAlpha);
   bool SutdownEGLLibraryIfNecessary();
   RefPtr<ID3D11Query> GetD3D11Query();
+  void ReleaseNativeCompositorResources();
 
   EGLConfig mEGLConfig;
   EGLSurface mEGLSurface;
@@ -126,8 +130,11 @@
   RenderedFrameId mLastCompletedFrameId;
 
   Maybe<LayoutDeviceIntSize> mBufferSize;
+  bool mUseNativeCompositor;
   bool mUsePartialPresent;
   bool mFullRender;
+  // Used to know a timing of disabling native compositor.
+  bool mDisablingNativeCompositor;
 };
 
 }  // namespace wr